src/regmod/utils.py

```diff
@@ -8,8 +8,32 @@
 from xspline import XSpline
 
 
-def default_vec_factory(vec: Any, size: int, default_value: float,
+def default_vec_factory(vec: Any,
+                        size: int,
+                        default_value: float,
                         vec_name: str = 'vec') -> np.ndarray:
+    """Validate or create the vector.
+
+    Parameters
+    ----------
+    vec : Any
+        Vector to be validated. When it is ``None``, it will use the
+        ``default_value`` and ``size`` to create the vector. When it is a
+        scalar, it will use ``size`` to create the vector. When it is a vector,
+        it will be checked if have correct size.
+    size : int
+        Size of the vector.
+    default_value : float
+        Default value of the vector.
+    vec_name : str, optional
+        Name of the vector, used for print error purpose. Default to ``'vec'``.
+
+    Returns
+    -------
+    np.ndarray
+        Created or validated vector.
+    """
+
     if vec is None:
         vec = np.repeat(default_value, size)
```

Comment:
You could also change if, elif, else to just if statements with returns inside

Reply:
Thanks very much simplify the syntax :)